File: XFile55.cpp
#include "pt4.h"
#include <fstream>
using namespace std;

void Solve()
{
    Task("XFile55");
    int N;
    string s0,s;
    pt >> s0;
	pt >> N;
	fstream* f = new fstream[N+1];
	for(int i = 0; i < N; ++i){
		pt >> s;
		f[i].open(s, ios::binary | ios::in);
	}
	f[N].open(s0, ios::binary | ios::out);
	///////////////////////////////////////////
	for(int i = 0; i < N; ++i){
		f[i].seekg(0);
		f[i].seekg(f[i].tellg());
		int num = 0;
		while (f[i].peek() != -1)
		{
			int x;
			f[i].read((char *)&x, sizeof(x));
			num++;
		}
		f[N].write((char *)&num, sizeof(num));
		num = 0;
		f[i].seekg(0);
		f[i].seekg(f[i].tellg());
		while (f[i].peek() != -1)
		{
			int x;
			f[i].read((char *)&x, sizeof(x));
			f[N].write((char *)&x, sizeof(x));
		}
	}
	
	///////////////////////////////////
	for (int i = 0; i <= N; ++i)
		f[i].close();
	delete[] f;
}

File: XFile55.cs
// File: "XFile55"
using PT4;
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PT4Tasks
{
    public class MyTask : PT
    {
        public static void Solve()
        {
            Task("XFile55");
            BinaryWriter bw = new BinaryWriter(File.OpenWrite(GetString()));
            int N = GetInt();
            BinaryReader br;
            int num = 0;
            for (int i = 0; i < N; i++)
            {
                br = new BinaryReader(File.OpenRead(GetString()));
                bw.Write(((int)(br.BaseStream.Length / sizeof(int))));
                while (br.BaseStream.Position < br.BaseStream.Length)
                {
                    num = br.ReadInt32();
                    Show(num);
                    bw.Write(num);
                }
                br.Close();
                br.Dispose();
            }
            bw.Close();
            bw.Dispose();
        }
    }
}
